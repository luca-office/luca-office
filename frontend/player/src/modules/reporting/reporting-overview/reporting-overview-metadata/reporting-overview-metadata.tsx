import {css} from "@emotion/react"
import {round} from "lodash"
import * as React from "react"
import {Heading, Paper, ProgressBar, Text} from "shared/components"
import {HeadingLevel} from "shared/enums"
import {
  backgroundColorBright,
  backgroundColorLight,
  borderRadius,
  borderRadiusHuge,
  CustomStyle,
  FontWeight,
  insetShadow,
  spacing,
  spacingCard,
  spacingLarge,
  spacingMedium,
  spacingSmaller,
  TextSize
} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {roundNumber} from "shared/utils"
import {ReportingOverviewSize, reportingOverviewStyles} from "../reporting-overview.style"

export interface ReportingOverviewMetadataProps extends CustomStyle {
  readonly averageScore: number
  readonly maximumScore: number
  readonly participantScore: number
}

export const ReportingOverviewMetadata: React.FC<ReportingOverviewMetadataProps> = ({
  customStyles,
  averageScore,
  maximumScore,
  participantScore
}) => {
  const {t} = useLucaTranslation()

  return (
    <div css={[reportingOverviewStyles.dataContentWrapper, customStyles]}>
      <Heading level={HeadingLevel.h3} fontWeight={FontWeight.Bold}>
        {t("reporting_overview_reached_points")}
      </Heading>
      <div css={styles.metadataContent}>
        <ProgressBar
          customStyles={styles.progressBar}
          customHeight={ReportingOverviewSize.progressBar}
          progressInPercent={roundNumber((participantScore / maximumScore) * 100)}
          verticalLinePositionInPercent={roundNumber((averageScore / maximumScore) * 100)}
          verticalLineTooltipText={`${t("dashboard__progress_reports_survey_average")} ${
            averageScore === 1
              ? t("coding_criteria__criterion_list_score_singular", {score: averageScore})
              : t("coding_criteria__criterion_list_score", {score: averageScore})
          }`}
        />
        <div css={styles.metadataContentPointsWrapper}>
          <div css={styles.metadataContentPoints}>
            <div css={reportingOverviewStyles.dataContentWrapper}>
              <Heading level={HeadingLevel.h3} fontWeight={FontWeight.Bold}>
                {t("reporting_overview_maximum_points")}
              </Heading>
              <Paper customStyles={styles.scorePaper}>
                {maximumScore === 1
                  ? t("coding_criteria__criterion_list_score_singular", {score: maximumScore})
                  : t("coding_criteria__criterion_list_score", {score: maximumScore})}
              </Paper>
            </div>
            <div css={reportingOverviewStyles.dataContentWrapper}>
              <Heading level={HeadingLevel.h3} fontWeight={FontWeight.Bold}>
                {t("reporting_result__average_finalscore_label")}
              </Heading>
              <Paper customStyles={styles.scorePaper}>
                {t("coding_criteria__criterion_list_score", {score: round(averageScore, 2)})}
              </Paper>
            </div>
          </div>
          <div css={reportingOverviewStyles.dataContentWrapper}>
            <Heading customStyles={styles.participantScoreLabel} level={HeadingLevel.h3} fontWeight={FontWeight.Bold}>
              {t("reporting__overview_participant_score")}
            </Heading>
            <Text size={TextSize.Larger}>{t("coding_criteria__criterion_list_score", {score: participantScore})}</Text>
          </div>
        </div>
      </div>
    </div>
  )
}

const styles = {
  metadataContent: css({
    display: "grid",
    gridTemplateRows: "repeat(2, minmax(min-content, max-content))",
    gridRowGap: spacingLarge,
    boxShadow: insetShadow,
    borderRadius: borderRadius,
    backgroundColor: backgroundColorLight,
    padding: spacingMedium,
    boxSizing: "border-box"
  }),
  metadataContentPointsWrapper: css({
    display: "grid",
    gridTemplateColumns: "1fr minmax(min-content, max-content)",
    gridColumnGap: spacingMedium
  }),
  metadataContentPoints: css({
    display: "grid",
    gridTemplateColumns: "repeat(2, minmax(min-content, max-content))",
    gridColumnGap: spacingMedium
  }),
  scorePaper: css({
    minWidth: ReportingOverviewSize.scorePaper,
    padding: spacing(spacingSmaller, spacingCard),
    backgroundColor: backgroundColorBright
  }),
  participantScoreLabel: css({
    textAlign: "end"
  }),
  progressBar: css({
    borderRadius: borderRadiusHuge
  })
}
