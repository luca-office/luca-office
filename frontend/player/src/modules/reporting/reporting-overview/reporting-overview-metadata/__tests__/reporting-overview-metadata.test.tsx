import * as React from "react"
import {create} from "react-test-renderer"
import {ReportingOverviewMetadata, ReportingOverviewMetadataProps} from "../reporting-overview-metadata"

const defaultProps: ReportingOverviewMetadataProps = {
  averageScore: 6,
  maximumScore: 12,
  participantScore: 8
}

const getComponent = (props?: Partial<ReportingOverviewMetadataProps>) => (
  <ReportingOverviewMetadata {...{...defaultProps, ...props}} />
)

describe("reporting-overview-metadata", () => {
  it("renders correctly", () => {
    const component = getComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
