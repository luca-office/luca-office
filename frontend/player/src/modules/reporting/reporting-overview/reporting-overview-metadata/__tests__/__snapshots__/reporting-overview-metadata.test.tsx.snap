// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reporting-overview-metadata renders correctly 1`] = `
<div
  className="css-1wz2obg-ReportingOverviewMetadata"
>
  <div
    className="css-8050ac-Heading"
  >
    reporting_overview_reached_points
  </div>
  <div
    className="css-os5ou9-ReportingOverviewMetadata"
  >
    <div
      className="css-10fw6iz-ProgressBar"
    >
      <div
        className="progress-element css-12lv16f-ProgressBar"
      />
      <div
        className="css-0"
        id="tooltip_42"
      >
        <div
          className="css-plc2dy-ProgressBar"
        />
      </div>
      <div
        className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
        role="tooltip"
        style={
          {
            "backgroundColor": "#2e3032",
            "zIndex": 1,
          }
        }
      >
        dashboard__progress_reports_survey_average coding_criteria__criterion_list_score
        <div
          className="react-tooltip-arrow styles-module_arrow__K0L3T"
          style={{}}
        />
      </div>
    </div>
    <div
      className="css-6rsjbu-ReportingOverviewMetadata"
    >
      <div
        className="css-mfdu1x-ReportingOverviewMetadata"
      >
        <div
          className="css-udnqo4-ReportingOverviewMetadata"
        >
          <div
            className="css-8050ac-Heading"
          >
            reporting_overview_maximum_points
          </div>
          <div>
            <div
              className="css-1dmxdyc-Paper"
            >
              coding_criteria__criterion_list_score
            </div>
          </div>
        </div>
        <div
          className="css-udnqo4-ReportingOverviewMetadata"
        >
          <div
            className="css-8050ac-Heading"
          >
            reporting_result__average_finalscore_label
          </div>
          <div>
            <div
              className="css-1dmxdyc-Paper"
            >
              coding_criteria__criterion_list_score
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-udnqo4-ReportingOverviewMetadata"
      >
        <div
          className="css-egp9h9-Heading"
        >
          reporting__overview_participant_score
        </div>
        <div
          className="css-1iyg2r7-Text"
        >
          coding_criteria__criterion_list_score
        </div>
      </div>
    </div>
  </div>
</div>
`;
